modules/carto/src/api/maps-v3-client.ts

```diff
@@ -224,10 +297,11 @@ async function _fetchDataUrl({
         break;
       }
     }
+    log.assert(url!, 'Unsupported data formats received from backend.');
   }
 
   const {accessToken} = localCreds;
-  return {url, accessToken, mapFormat, metadata};
+  return {url: url!, accessToken, mapFormat: mapFormat!, metadata};
```

Comment:
General note w.r.t migration to Typescript.

We use log.assert extensively and we expect that it throws. We know it, but with current typings, typescript don't, so we have to "teach it" using ! or other ugly casts.

Best way, would be to tell that log.assert throws if url is non-thruthy - [it's possible](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions).

This would require either our own local assert function with signature like this:

```
function assert(condition, message) asserts condition {}
```
Or update deck.gl to properly export log object ... only after patching probe.gl.

Reply:
Now the only problem with using `assert` is that asserts are typically used for "development mode/time" checks and frequently, there is expectation that in production builds they are disabled and/or completly removed from code. They are in several runtimes/libraries.

Question is whether deck.gl/probe.gl wants to rely on asserts as normal runtime checks or maybe there should be new concept of `runtimeAssert` that has same semantics but won't be removed by some future tooling.

Reply:
For record: https://github.com/uber-web/probe.gl/pull/188

Reply:
Resolved temporarily for carto module by introducing carto-module local `assert` helper that has correct typings.